### PR TITLE
CRM-20007 Add test to prove that repeatTransaction does not use the P…

### PR DESCRIPTION
**Pull request by seamuslee001**
…ayment Procesors Payment Instrument when creating contributions

### CODE REVIEW

**Comment by seamuslee001** [comment]
ping @eileenmcnaughton I think this is what you were looking for right?

File: tests/phpunit/api/v3/ContributionTest.php [hunk 1]
```diff
@@ -3323,4 +3323,33 @@ protected function setUpForCompleteTransaction() {
...
+        $this->_params,
+        array('contribution_recur_id' => $contributionRecur['id']))
+    );
+    $paymentInstruments = CRM_Contribute_PseudoConstant::paymentInstrument('name');
+    $contribution2 = $this->callAPISuccess('contribution', 'repeattransaction', array(
+      'contribution_status_id' => 'Completed',
+      'trxn_id' => uniqid(),
+      'original_contribution_id' => $contribution1,
+    ));
+    $this->assertEquals(array_search('Debit Card', $paymentInstruments), $contribution2['values'][$contribution2['id']]['payment_instrument_id']);
```
Comment (seamuslee001) [inline tests/phpunit/api/v3/ContributionTest.php]: That is what is used in the PaymentProcessor create in the test suite

**Comment by seamuslee001** [comment]
pinging @andrew-cormick-dockery @ineffyble @mrlavalava  You folks may be interested in this.

**Comment by eileenmcnaughton** [comment]
I've taken a look at this & agree that the test replicates a real issue & the fix is appropriate

**Pull request merged**